spec/integration/spec_helper.cr

```diff
@@ -0,0 +1,172 @@
+  create_git_release "mock", "0.1.0", "name: mock\nversion: 0.1.0\n" +
+                                      "dependencies:\n  shoulda:\n    git: #{git_path("shoulda")}\n    version: < 0.3.0\n" +
+                                      "development_dependencies:\n  minitest:\n    git: #{git_path("minitest")}\n"
+
+  create_git_repository "orm", "0.1.0", "0.2.0", "0.3.0", "0.3.1", "0.3.2", "0.4.0"
+  create_git_release "orm", "0.5.0", "name: orm\nversion: 0.5.0\ndependencies:\n  pg:\n    git: #{git_path("pg")}\n    version: < 0.3.0\n"
+
+  create_git_repository "release", "0.2.0", "0.2.1", "0.2.2"
+  create_git_release "release", "0.3.0", "name: release\nversion: 0.3.0\ncustom_dependencies:\n  pg:\n    git: #{git_path("optional")}\n"
+
+  # git dependencies with prereleases:
+  create_git_repository "unstable", "0.1.0", "0.2.0", "0.3.0.alpha", "0.3.0.beta"
+  create_git_repository "preview", "0.1.0", "0.2.0", "0.3.0.a", "0.3.0.b", "0.3.0", "0.4.0.a"
+
+  # path dependency:
+  create_path_repository "foo", "0.1.0"
+
+  # dependency with neither a shard.yml and/or version tags:
+  # create_git_repository "empty"
+  # create_git_commit "empty", "initial release"
+
+  create_git_repository "missing"
+  create_shard "missing", "name: missing\nversion: 0.1.0\n"
+  create_git_commit "missing", "initial release"
+
+  # dependencies with postinstall scripts:
+  create_git_repository "post"
+  create_file "post", "Makefile", "all:\n\ttouch made.txt\n"
+  create_git_release "post", "0.1.0", "name: post\nversion: 0.1.0\nscripts:\n  postinstall: make\n"
+
+  create_git_repository "fails"
+  create_file "fails", "Makefile", "all:\n\ttest -n ''\n"
+  create_git_release "fails", "0.1.0", "name: fails\nversion: 0.1.0\nscripts:\n  postinstall: make\n"
+
+  # transitive dependencies in postinstall scripts:
+  create_git_repository "version"
+  create_file "version", "src/version.cr", %(module Version; STRING = "version @ 0.1.0"; end)
+  create_git_release "version", "0.1.0"
+
+  create_git_repository "transitive"
+  create_file "transitive", "src/version.cr", %(require "version"; puts Version::STRING)
+  create_git_release "transitive", "0.2.0", <<-YAML
+    name: transitive
+    version: 0.2.0
+    dependencies:
+      version:
+        git: #{git_path(:version)}
+    scripts:
+      postinstall: crystal build src/version.cr
+    YAML
+
+  # dependencies with executables:
+  create_git_repository "binary"
+  create_file "binary", "bin/foobar", "#! /usr/bin/env sh\necho 'OK'", perm: 0o755
+  create_file "binary", "bin/baz", "#! /usr/bin/env sh\necho 'KO'", perm: 0o755
+  create_git_release "binary", "0.1.0", "name: binary\nversion: 0.1.0\nexecutables:\n  - foobar\n  - baz\n"
+  create_file "binary", "bin/foo", "echo 'FOO'", perm: 0o755
+  create_git_release "binary", "0.2.0", "name: binary\nversion: 0.2.0\nexecutables:\n  - foobar\n  - baz\n  - foo"
+
+  Shards::Specs.created_repositories!
+end
+
+private def assert(value, message, file, line)
```

Comment:
Could we replace this with a matcher overload that accepts an `message` argument? That would be a good enhancement for the stdlb spec (for all matchers). We could already add it right here and only need to remove the custom implementation once it's landed in stdlib, not touch the actual specs again.
`#should` already supports `file` and `line` arguments.

Reply:
The custom message story was raised somewhere. I made the helper private so it does not leak outside of this file. The custom messages in this file added some value so that's why I didn't want to loose it.

If that feature lands in the std-lib we can iterate the specs for sure. I see no need to do it right away here.

Reply:
Oh, I missed the private visibility. It's fine if this is only used in the same file.